packages/components/src/internal/components/ontology/OntologyConceptAnnotation.tsx

```diff
@@ -106,14 +102,17 @@ export const OntologyConceptAnnotationImpl: FC<OntologyConceptAnnotationImplProp
                                 )}
                                 <td className="content">
                                     <a
-                                        className="domain-validator-link domain-annotation-item"
-                                        onClick={toggleConceptModal}
+                                        className={classNames("domain-annotation-item", isFieldLocked ? "domain-text-label" : "domain-validator-link")}
+                                        onClick={isFieldLocked ? null : toggleSelectModal}
```

Comment:
I pulled this up in the UI. It looks really good. I like the popover/hover details better and having the click on the current concept open the selector dialog. You don't have to do it as part of this PR, but now that the "initial concept" can be passed in to the selector, we could do that in this case.